### PR TITLE
Issue#2426 refactor breakup js files

### PR DESCRIPTION
**Pull request by adamjarling**
This PR mainly re-organizes the MEJS4 JS files, breaks out smaller classes for better structure and easier to navigate.

### CODE REVIEW

**Comment by cjcolvar** [comment]
Now that I merged #2476, looks like you need to update this PR to reflect the changes there.

**Comment by adamjarling** [comment]
Ah, got it.  Just fixed the conflict and re-pushed.  Thanks for the heads up.

File: app/assets/stylesheets/mejs4/mejs4_add_marker_to_playlist.scss [hunk 1]
```diff
@@ -8,6 +8,10 @@
   width: 20px;
 }
 
+.mejs__time-marker {
```
Comment (cjcolvar) [inline app/assets/stylesheets/mejs4/mejs4_add_marker_to_playlist.scss]: Why is this needed in the refactor?

Comment (adamjarling) [reply app/assets/stylesheets/mejs4/mejs4_add_marker_to_playlist.scss]: Consider this an enhancement to the refactor.   It was a small style bug most likely introduced when I committed the track scrubber code, but which I never caught.  And only happens when I call a method on the MEJS4 core markers plugin directly, to "re-paint" the markers after an update.

**Comment by adamjarling** [comment]
Thanks @cjcolvar .  An added benefit of re-structuring this way will make unit testing (when we introduce) much more manageable.
